### ISSUE TITLE
Weird SDK crash when collecting coverage from a test (on Linux)

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/sdk#49384]
https://github.com/google/mono_repo.dart/runs/7154777092?check_suite_focus=true#step:7:8462

```

===== CRASH =====
si_signo=Segmentation fault(11), si_code=1, si_addr=0x18
version=2.17.5 (stable) (Tue Jun 21 11:05:10 2022 +0200) on "linux_x64"
pid=1726, thread=1792, isolate_group=main(0x55911de40000), isolate=main(0x55911de40800)
isolate_instructions=55911a411460, vm_instructions=55911a411460
  pc 0x000055911aabec9e fp 0x00007f3351bfd1f0 dart::kernel::ScopeBuilder::LookupVariable(long)+0x1fe
  pc 0x000055911aabea5f fp 0x00007f3351bfd210 dart::kernel::ScopeBuilder::VisitVariableGet(long)+0xf
  pc 0x000055911aabc9b0 fp 0x00007f3351bfd250 dart::kernel::ScopeBuilder::VisitExpression()+0x610
  pc 0x000055911aabd308 fp 0x00007f3351bfd320 dart::kernel::ScopeBuilder::VisitConstructor()+0x228
  pc 0x000055911aabb3ed fp 0x00007f3351bfd4a0 dart::kernel::ScopeBuilder::BuildScopes()+0x97d
  pc 0x000055911a6730e7 fp 0x00007f3351bfd7e0 dart::ParsedFunction::EnsureKernelScopes()+0x37
  pc 0x000055911aa84c53 fp 0x00007f3351bfd810 dart::kernel::StreamingFlowGraphBuilder::ParseKernelASTFunction()+0x53
  pc 0x000055911aa84ab1 fp 0x00007f3351bfd8c0 dart::kernel::StreamingFlowGraphBuilder::BuildGraph()+0xb1
  pc 0x000055911aa98725 fp 0x00007f3351bfdb80 dart::kernel::FlowGraphBuilder::BuildGraph()+0x65
  pc 0x000055911a70e0e0 fp 0x00007f3351bfddc0 dart::DartCompilationPipeline::BuildFlowGraph(dart::Zone*, dart::ParsedFunction*, dart::ZoneGrowableArray<dart::ICData const*>*, long, bool)+0x40
  pc 0x000055911a70f102 fp 0x00007f3351bfe4d0 dart::CompileParsedFunctionHelper::Compile(dart::CompilationPipeline*)+0x4d2
  pc 0x000055911a70fd7c fp 0x00007f3351bfe6e0 /opt/hostedtoolcache/dart-sdk/bin/dart+0x21e3d7c
  pc 0x000055911a710165 fp 0x00007f3351bfe720 dart::Compiler::EnsureUnoptimizedCode(dart::Thread*, dart::Function const&)+0xb5
  pc 0x000055911a6e9a54 fp 0x00007f3351bfe770 dart::SourceReport::VisitFunction(dart::JSONArray*, dart::Function const&)+0x134
  pc 0x000055911a6e9eee fp 0x00007f3351bfe800 dart::SourceReport::VisitLibrary(dart::JSONArray*, dart::Library const&)+0x12e
  pc 0x000055911a6ea37f fp 0x00007f3351bfe8a0 dart::SourceReport::PrintJSON(dart::JSONStream*, dart::Script const&, dart::TokenPosition, dart::TokenPosition)+0x12f
  pc 0x000055911a6da55f fp 0x00007f3351bfe9f0 /opt/hostedtoolcache/dart-sdk/bin/dart+0x21ae55f
  pc 0x000055911a6d12a1 fp 0x00007f3351bfeb40 dart::Service::InvokeMethod(dart::Isolate*, dart::Array const&, bool)+0x321
  pc 0x000055911a6d19f1 fp 0x00007f3351bfeb60 dart::Service::HandleIsolateMessage(dart::Isolate*, dart::Array const&)+0x11
  pc 0x000055911a5b25e8 fp 0x00007f3351bfec40 dart::IsolateMessageHandler::HandleMessage(std::__2::unique_ptr<dart::Message, std::__2::default_delete<dart::Message> >)+0x278
  pc 0x000055911a5db54d fp 0x00007f3351bfecb0 dart::MessageHandler::HandleMessages(dart::MonitorLocker*, bool, bool)+0x14d
  pc 0x000055911a5dbb70 fp 0x00007f3351bfed00 dart::MessageHandler::TaskCallback()+0x120
  pc 0x000055911a6fd0b8 fp 0x00007f3351bfed80 dart::ThreadPool::WorkerLoop(dart::ThreadPool::Worker*)+0x148
  pc 0x000055911a6fd50d fp 0x00007f3351bfedb0 dart::ThreadPool::Worker::Main(unsigned long)+0x6d
  pc 0x000055911a672368 fp 0x00007f3351bfee70 /opt/hostedtoolcache/dart-sdk/bin/dart+0x2146368
-- End of DumpStackTrace
=== Crash occured when compiling package:mono_repo/src/commands/github/action_info.dart_ActionInfo_ActionInfo. in unoptimized JIT mode in unknown pass
=== Flow Graph not available
getVM: (112) Service has disappeared
package:vm_service/src/vm_service.dart 1746:45                                                     new _OutstandingRequest
package:vm_service/src/vm_service.dart 2262:21                                                     VmService._call
package:vm_service/src/vm_service.dart 2068:25                                                     VmService.getVM
package:coverage/src/collect.dart 191:28                                                           _resumeIsolates
package:coverage/src/collect.dart 98:13                                                            collect
===== asynchronous gap ===========================
dart:async                                                                                         _Completer.completeError
package:vm_service/src/vm_service.dart 1753:18                                                     _OutstandingRequest.completeError
package:vm_service/src/vm_service.dart 2347:15                                                     VmService._processResponse
package:vm_service/src/vm_service.dart 2336:7                                                      VmService._processMessageStr
package:vm_service/src/vm_service.dart 2288:7                                                      VmService._processMessage
===== asynchronous gap ===========================
dart:async                                                                                         _StreamImpl.listen
package:vm_service/src/vm_service.dart 1788:27                                                     new VmService
package:coverage/src/collect.dart 76:19                                                            collect.<fn>
===== asynchronous gap ===========================
dart:async                                                                                         _asyncThenWrapperHelper
package:coverage/src/collect.dart                                                                  collect.<fn>
package:coverage/src/util.dart 33:23                                                               retry.<fn>
package:coverage/src/util.dart 30:22                                                               retry.<fn>
package:coverage/src/util.dart 18:15                                                               retry.withTimeout
package:coverage/src/util.dart 30:10                                                               retry
package:coverage/src/collect.dart 67:11                                                            collect
/home/runner/.pub-cache/hosted/pub.dartlang.org/coverage-1.5.0/bin/collect_coverage.dart 22:28     main.<fn>
/home/runner/.pub-cache/hosted/pub.dartlang.org/coverage-1.5.0/bin/collect_coverage.dart 21:23     main.<fn>
package:stack_trace                                                                                Chain.capture
/home/runner/.pub-cache/hosted/pub.dartlang.org/coverage-1.5.0/bin/collect_coverage.dart 21:15     main
/home/runner/.pub-cache/hosted/pub.dartlang.org/coverage-1.5.0/bin/test_with_coverage.dart 198:26  main
```

`Dart SDK version: 2.17.5 (stable) (Tue Jun 21 11:05:10 2022 +0200) on "linux_x64"`

**Comment by kevmoo** [comment]
But the crash DOESN'T seem to happen on `dev` – 🤷  – will close out.

**Comment by KOGI** [comment]
This may not be happening on `dev`, but it's still happening on `2.17.6` which was tagged on Jul 13. Since that version is still the latest and only stable release, this seems like an issue that still needs to be addressed.

By upgrading my sdk, now all my tests fail and I'm unable to release without downgrading code.

**Comment by kevmoo** [comment]
@itsjustkevin – wondering if we want to consider a patch/cherry-pick?

**Comment by itsjustkevin** [comment]
@kevmoo labeling this as a cherry-pick candidate. 

A few things:
1. If this is not an issue on beta, we could cherry-pick the change into a hotfix. 
2. There doesn't seem to be a lot of users calling out this issue. 

Do you see this as a high priority issue?

**Comment by kevmoo** [comment]
@itsjustkevin – given we've only seen one user affected, I don't think this meets the bar for a hot fix for 2.17 – since we're locking down the next release relatively soon – sorry @KOGI
